proto/stream_plan.proto

```diff
@@ -186,10 +186,11 @@ message SinkNode {
 message ProjectNode {
   repeated expr.ExprNode select_list = 1;
   // this two field is expressing a list of usize pair, which means when project receives a
-  // watermark with `watermark_input_key[i]` column index, it should derive a new watermark
-  // with `watermark_output_key[i]`th expression
-  repeated uint32 watermark_input_key = 2;
-  repeated uint32 watermark_output_key = 3;
+  // watermark with `watermark_input_cols[i]` column index, it should derive a new watermark
+  // with `watermark_output_cols[i]`th expression
+  repeated uint32 watermark_input_cols = 2;
+  repeated uint32 watermark_output_cols = 3;
+  repeated uint32 nondecreasing_exprs = 4;
```

Comment:
This will not become a breaking change, right? Because the `nondecreasing_exprs` will always be empty for `Projects`, except the ones created from generated column with `proctime`

Reply:
Btw, how about simply naming it as `watermark_exprs`? It sounds more straight-forward to me

Reply:
> This will not become a breaking change, right? Because the `nondecreasing_exprs` will always be empty for Projects, except the ones created from generated column with `proctime`

Yep.

> Btw, how about simply naming it as `watermark_exprs`?

I think it's a little bit ambiguous cuz the word is also used in `WatermarkFilter` representing the watermark delay expression. Also it can't clearly imply the watermark is produced by `Project`. What about just put these indices at the end of `watermark_output_cols`? When creating executor, build a vec called `watermark_cols_to_generate` from them, accordingly change the previous name `watermark_derivations` to `watermark_cols_to_derive`.

Reply:
> When creating executor, build a vec called watermark_cols_to_generate from them, accordingly change the previous name watermark_derivations: MultiMap<usize, usize> to watermark_cols_to_derive.

This sounds better to me.